lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/mic.rb

```diff
@@ -0,0 +1,196 @@
+              'mic_stop'  => [ 'audio_mic_stop' ],
+              'mic_list'  => [ 'audio_mic_list' ],
+              'listen'    => [ 'audio_mic_start' ]
+            }
+
+            filter_commands(all, reqs)
+          end
+
+          #
+          # Name for this dispatcher
+          #
+          def name
+            "Stdapi: Mic"
+          end
+
+          def cmd_mic_list
+            client.mic.mic_list
+            if client.mic.mic_list.length == 0
+              print_error("No mics were found")
+              return
+            end
+
+            client.mic.mic_list.each_with_index do |name, indx|
+              print_line("#{indx + 1}: #{name}")
+            end
+          end
+
+          def audio_file_wave_header(sample_rate_hz, num_channels, bits_per_sample, data_size)
+            subchunk1_size = 16
+            chunk_size = 4 + (8 + subchunk1_size) + (8 + data_size)
+            byte_rate = sample_rate_hz * num_channels * bits_per_sample / 8
+            block_align = num_channels * bits_per_sample / 8
+
+            [
+              BinData::Int32be.new(0x52494646),      # ChunkID: "RIFF"
+              BinData::Int32le.new(chunk_size),      # ChunkSize
+              BinData::Int32be.new(0x57415645),      # Format: "WAVE"
+              BinData::Int32be.new(0x666d7420),      # SubChunk1ID: "fmt "
+              BinData::Int32le.new(16),              # SubChunk1Size
+              BinData::Int16le.new(1),               # AudioFormat
+              BinData::Int16le.new(num_channels),    # NumChannels
+              BinData::Int32le.new(sample_rate_hz),  # SampleRate
+              BinData::Int32le.new(byte_rate),       # ByteRate
+              BinData::Int16le.new(block_align),     # BlockAlign
+              BinData::Int16le.new(bits_per_sample), # BitsPerSample
+              BinData::Int32be.new(0x64617461),      # SubChunk2ID: "data"
+              BinData::Int32le.new(data_size)        # SubChunk2Size
+            ]
+          end
+
+          def cmd_mic_start(*args)
+            get_data = lambda do |channel, file|
+              data = channel.read(65536)
+              if data
+                ::File.open(file, 'a') do |f|
+                  f.write(data)
+                end
+                return data.length
+              end
+              return 0
+            end
+            device_id = 1
+            duration = 1800
+            saved_audio_path = Rex::Text.rand_text_alpha(8) + ".wav"
+
+            mic_start_opts = Rex::Parser::Arguments.new(
+              "-h" => [ false, "Help Banner" ],
+              "-d" => [ true, "The stream duration in seconds (Default: 1800)" ], # 30 min
+              "-m" => [ true, "Microphone device index to record from (1: system default)" ],
+              "-s" => [ true, "The saved audio file path (Default: '#{saved_audio_path}')" ]
+            )
+
+            mic_start_opts.parse(args) do |opt, _idx, val|
+              case opt
+                when "-h"
```

Comment:
I know Dave disagrees with me here, but the style guide sides with me that `when` should be indented at the same level as `case`. Same for the rest of these.